process/scToProtocol/stakingToPeer_test.go

```diff
@@ -0,0 +1,661 @@
+	return block.Body{
+		{
+			TxHashes:        [][]byte{[]byte("hash1"), []byte("hash2")},
+			ReceiverShardID: 0,
+			SenderShardID:   0,
+			Type:            block.SmartContractResultBlock,
+		},
+	}
+}
+
+func TestNewStakingToPeerNilAddrConverterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.AdrConv = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilAddressConverter, err)
+}
+
+func TestNewStakingToPeerNilHasherShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.Hasher = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilHasher, err)
+}
+
+func TestNewStakingToPeerNilMarshalizerShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.Marshalizer = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilMarshalizer, err)
+}
+
+func TestNewStakingToPeerNilPeerAccountAdapterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.PeerState = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilPeerAccountsAdapter, err)
+}
+
+func TestNewStakingToPeerNilBaseAccountAdapterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.BaseState = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilAccountsAdapter, err)
+}
+
+func TestNewStakingToPeerNilArgumentParserShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.ArgParser = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilArgumentParser, err)
+}
+
+func TestNewStakingToPeerNilCurrentBlockHeaderShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.CurrTxs = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilTxForCurrentBlockHandler, err)
+}
+
+func TestNewStakingToPeerNilScDataGetterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.ScDataGetter = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilSCDataGetter, err)
+}
+
+func TestNewStakingToPeer_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+
+	stakingToPeer, err := NewStakingToPeer(arguments)
+	assert.NotNil(t, stakingToPeer)
+	assert.Nil(t, err)
+}
+
+func TestStakingToPeer_UpdateProtocolCannotGetTxShouldErr(t *testing.T) {
+	t.Parallel()
+
+	testError := errors.New("error")
+	currTx := &mock.TxForCurrentBlockStub{}
+	currTx.GetTxCalled = func(txHash []byte) (handler data.TransactionHandler, e error) {
+		return nil, testError
+	}
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.CurrTxs = currTx
+	blockBody := createBlockBody()
```

Comment:
you can move this line with `blockBody` right before you use it (line 150). This can be applied in all the tests

Reply:
Done.